framework/validators/EmailValidator.php

```diff
@@ -111,21 +112,24 @@ protected function validateValue($value)
      */
     protected function isDNSValid($domain)
     {
-        if (checkdnsrr($domain . '.', 'MX')) {
-            $mxRecords = dns_get_record($domain . '.', DNS_MX);
-            if ($mxRecords !== false && count($mxRecords) > 0) {
-                return true;
-            }
+        return $this->hasDNSRecord($domain, true) || $this->hasDNSRecord($domain, false);
+    }
+
+    private function hasDNSRecord($domain, $mx)
```

Comment:
```suggestion
    private function hasDNSRecord($domain, $isMx)
```

Reply:
ok

Reply:
I suggests: You can directly transfer a DNS record types to it.
```php
private function hasDnsRecord($domain, $dnsRecordTypes = DNS_MX);
```

Reply:
It might be better to get all the types you want in a single step (one DNS request), including the ipv6 address, e.g. `DNS_A | DNS_AAAA | DNS_MX`.

It might be enough to get the name servers and MX: `DNS_NS | DNS_MX`